packages/permissions/src/hooks/use-is-authorized/use-is-authorized.ts

```diff
@@ -85,23 +114,11 @@ const useIsAuthorized = ({
     )}.`
   );
 
-  const actualPermissions =
-    useApplicationContext<TNormalizedPermissions | null>(
-      (applicationContext) =>
-        projectPermissions?.permissions ?? applicationContext.permissions
-    );
-  const actualActionRights =
-    useApplicationContext<TNormalizedActionRights | null>(
-      (applicationContext) =>
-        projectPermissions?.actionRights ?? applicationContext.actionRights
-    );
-  const actualDataFences = useApplicationContext<TNormalizedDataFences | null>(
-    (applicationContext) =>
-      projectPermissions?.dataFences ?? applicationContext.dataFences
-  );
+  const { resolvedPermissions, resolvedActionRights, resolvedDataFences } =
```

Comment:
Up until now we were reading the permissions from the application context but this is not enough when using this hook from a Custom View, since we won't have them over there but in the custom view context so I decided to create a private utility function to try to fetch permissions from both so we don't need to change the signature of this hook or create a new one.

Reply:
I'm a bit confused. Isn't the `ApplicationContext` also used in Custom Views? The permissions are usually exposed from the `FetchProject` data and mapped into the application context.

I'm also wondering if we really need a separate context for Custom Views but (re)use the Application Context and inject some `customView` specific data.

Let's have a chat about this maybe.

Reply:
Yes, that change wasn't actually needed. I implemented it at some point where things were not working but I think I then solved it the right way.
I rolled back that change in this commit: https://github.com/commercetools/merchant-center-application-kit/commit/f4caaf5d75231242b6f6a73764f2317c43f1ceeb

Regarding unifying contexts, I still believe is better for organization to have different ones serving different purposes.
Also, the new one would not change at runtime because it's for improve context data to be easily available throughout the Custom View code and it will be populated with the (static) data received from the host application.